packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx

```diff
@@ -0,0 +1,542 @@
+  )
+
+  const columns = useMemo<ColumnDef<EntityHeaderOrDummy, any>[]>(
+    () =>
+      isEditable
+        ? selectColumns.concat(entityHeaderColumns)
+        : entityHeaderColumns,
+    [entityHeaderColumns, isEditable, selectColumns],
+  )
+  const selectionCount = Object.keys(rowSelection).length
+  const {
+    data: results,
+    isSuccess,
+    isLoading,
+  } = useGetEntityHeaders(refsInState, {
+    useErrorBoundary: true,
+  })
+
+  const data = useMemo(() => {
+    //create dummy entries for values that were not returned by the getEntityHeaders call!
+    const newData = results ? results?.results : []
+    const newDataEntityIds = new Set()
+    newData.map(entityHeader => newDataEntityIds.add(entityHeader.id))
+    const missingRefs = refsInState.filter(
+      ref => !newDataEntityIds.has(ref.targetId),
+    )
+    const dummyEntityHeaders: EntityHeaderOrDummy[] = missingRefs.map(ref => {
+      return {
+        id: ref.targetId,
+        name: ref.targetId,
+        benefactorId: -1,
+        type: 'org.sagebionetworks.repo.model.Project',
+        createdOn: '',
+        modifiedOn: '',
+        createdBy: '',
+        modifiedBy: '',
+        isLatestVersion: true,
+        isDummy: true,
+      }
+    })
+    return newData.concat(dummyEntityHeaders)
+  }, [results])
+  const table = useReactTable({
+    data,
+    columns,
+    state: {
+      rowSelection,
+      columnFilters,
+    },
+    enableRowSelection: isEditable,
+    onRowSelectionChange: setRowSelection,
+    onColumnFiltersChange: setColumnFilters,
+    getCoreRowModel: getCoreRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFacetedRowModel: getFacetedRowModel(),
+    getFacetedUniqueValues: getFacetedUniqueValues(),
+    getFacetedMinMaxValues: getFacetedMinMaxValues(),
+    // debugTable: true,
+    // debugHeaders: true,
+    // debugColumns: false,
+    columnResizeMode: 'onChange',
+  })
+
+  if (isLoading) {
+    return <SkeletonTable numCols={3} numRows={10} />
+  } else if (!isSuccess) {
+    return <></>
+  }
+  const onRemoveFromAR: MouseEventHandler<HTMLButtonElement> = e => {
+    // rowSelection looks like {3: true. 5: true} where the key is the row index.
+    // Create a new ReferenceList based on the entityHeaders in the current table.
+    const updatedData = data.filter(
+      (_value, index) => !(rowSelection[index] === true),
+    )
+    const newRowRefs: ReferenceList = updatedData.map(entityHeader => {
+      return {
+        targetId: entityHeader.id,
+      }
+    })
+    updateRefsInState(newRowRefs)
+  }
+  const selectableTypes = [
+    EntityType.PROJECT,
+    EntityType.FOLDER,
+    EntityType.FILE,
+    EntityType.TABLE,
+    EntityType.LINK,
+    EntityType.ENTITY_VIEW,
+    EntityType.DOCKER_REPO,
+    EntityType.SUBMISSION_VIEW,
+    EntityType.DATASET,
+    EntityType.DATASET_COLLECTION,
+    EntityType.MATERIALIZED_VIEW,
+    EntityType.VIRTUAL_TABLE,
+  ]
```

Comment:
If this is everything, and it's only used for the entity finder, you can delete this. The default value is all types.

If it's used for something other than the EntityFinder, you can use `Object.values(EntityType)`, or consider adding a new entry `'ALL_TYPES'` to `EntityTypeGroup` in `EntityTypeUtils`.